lib/Serialization/ModuleFileSharedCore.cpp

```diff
@@ -303,6 +303,35 @@ validateControlBlock(llvm::BitstreamCursor &cursor,
       result.sdkName = blobData;
       break;
     }
+    case control_block::REVISION: {
+      // Tagged compilers should load only resilient modules if they were
+      // produced by the exact same version.
+
+      // Disable this restriction for compiler testing by setting this
+      // env var to any value.
+      static const char* ignoreRevision =
+        ::getenv("SWIFT_DEBUG_IGNORE_SWIFTMODULE_REVISION");
+      if (ignoreRevision)
+        break;
+
+      // Override this env var for testing, forcing the behavior of a tagged
+      // compiler and using the env var value to override this compiler's
+      // revision.
+      static const char* forcedDebugRevision =
+        ::getenv("SWIFT_DEBUG_FORCE_SWIFTMODULE_REVISION");
+
+      bool isCompilerTagged = forcedDebugRevision ||
+-        !version::Version::getCurrentCompilerVersion().empty();
+
+      StringRef moduleRevision = blobData;
+      if (isCompilerTagged && !moduleRevision.empty()) {
+        std::string compilerRevision = forcedDebugRevision ?
```

Comment:
`getSwiftRevision` could return a StringRef instead, in which case this could be a StringRef as well

Reply:
Good point, I've updated the function.